src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -58,7 +58,7 @@ public void AddReferencedCollection(string collection, string referencedCollecti
         public dynamic Id(dynamic doc)
         {
             var json = (DynamicBlittableJson)doc;
```

Comment:
If `doc` is `DynamicNullObject`, your cast will fail, need to do the check higher than this.

Reply:
Done